fsthttp/backend.go

```diff
@@ -245,40 +246,59 @@ func (b *BackendOptions) UseSSL(v bool) *BackendOptions {
 
 // SSLMinVersion sets the minimum allowed TLS version on SSL connections to this backend.
 func (b *BackendOptions) SSLMinVersion(min TLSVersion) *BackendOptions {
+	b.abiOpts.UseSSL(true)
 	b.abiOpts.SSLMinVersion(fastly.TLSVersion(min))
 	return b
 }
 
 // SSLMaxVersion sets the maximum allowed TLS version on SSL connections to this backend.
 func (b *BackendOptions) SSLMaxVersion(max TLSVersion) *BackendOptions {
+	b.abiOpts.UseSSL(true)
 	b.abiOpts.SSLMaxVersion(fastly.TLSVersion(max))
 	return b
 }
 
 // CertHostname sets the hostname that the server certificate should declare.
 func (b *BackendOptions) CertHostname(host string) *BackendOptions {
+	b.abiOpts.UseSSL(true)
 	b.abiOpts.CertHostname(host)
 	return b
 }
 
 // CACert sets the CA certificate to use when checking the validity of the backend.
 func (b *BackendOptions) CACert(cert string) *BackendOptions {
+	b.abiOpts.UseSSL(true)
 	b.abiOpts.CACert(cert)
 	return b
 }
 
 // Ciphers sets the list of OpenSSL ciphers to support for connections to this origin.
 func (b *BackendOptions) Ciphers(ciphers string) *BackendOptions {
+	b.abiOpts.UseSSL(true)
 	b.abiOpts.Ciphers(ciphers)
 	return b
 }
 
 // SNIHostname sets the SNI hostname to use on connections to this backend.
 func (b *BackendOptions) SNIHostname(host string) *BackendOptions {
+	b.abiOpts.UseSSL(true)
 	b.abiOpts.SNIHostname(host)
 	return b
 }
 
+// ClientCertificate sets the client certificate to be provided to the server as part of the SSL handshake.
```

Comment:
```suggestion
// ClientCertificate sets the client certificate to be provided to the server as part of the SSL handshake.
// Setting this will enable SSL for the connection as a side effect.
```